Signal/src/UIView+OWS.m

```diff
@@ -100,14 +100,19 @@ - (void)autoPinHeightToHeightOfView:(UIView *)view
 
 - (NSLayoutConstraint *)autoPinToSquareAspectRatio
 {
+    return [self autoPinToAspectRatio:1.0];
+}
+
+- (NSLayoutConstraint *)autoPinToAspectRatio:(CGFloat)ratio
+{
     self.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
OWSAssert(ration > 0.f);
OWSAssert(ration < 1.f);

Reply:
We also might want to clamp this value so that it fails gracefully in production, e.g.

```
// Clamp to ensure view has reasonable aspect ratio.
ratio = MAX(0.05f, (MIN(0.95f, ratio));
```

Reply:
Good call! Though we'd want a much large max ratio than .95 right? e.g. maybe something like 95.0?

Reply:
dur, you're right.  
```
// Clamp to ensure view has reasonable aspect ratio.
static const CGFloat kMinRatio = 0.05f;
ratio = MAX(kMinRatio, (MIN(1 / kMinRatio, ratio));
```